lisa/notifiers/text_result.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Any, List, Type, Union
+from pathlib import Path
+
+from lisa import notifier, schema
+from lisa.runner import print_results
+from lisa.testsuite import TestResult, TestResultMessage
+from lisa.util import LisaException
+
+
+class TextResult(notifier.Notifier):
+    """
+    Creating log notifier to dump text formatted results for easier
+    view in editing mode. The original log is complete but too long to
+    check only the summary.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "text_result"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _received_message(self, message: TestResultMessage) -> None:
+        if isinstance(message, TestResultMessage):
+            if message.is_completed:
+                self.received_messages.append(Union(None, message))
+        else:
+            raise LisaException("Received unsubscribed message type")
+
+    def _subscribed_message_type(self) -> List[Type[notifier.MessageBase]]:
+        return [TestResultMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self.received_messages: List[Union[TestResult, TestResultMessage]] = []
+
+    def finalize(self) -> None:
+        result_path = Path(
```

Comment:
This check should happen in initialize, not finalize. Let the file exists error happens earlier, instead of at end of run.